### ISSUE TITLE
[Bug]: search failed with error `node offline[node=16]: channel not available[channel=xxx]` or `failed to get delegator 1 for channel xxx: can not find client of node 1` after recover from querynode pod kill chaos test

### ISSUE THREAD

**Issue by zhuwenxing** [milvus-io/milvus#31029]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version:master-20240304-52540fec-amd64
- Deployment mode(standalone or cluster):cluster
- MQ type(rocksmq, pulsar or kafka):    
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

```

[2024-03-04T19:56:12.726Z] [2024-03-04 19:56:11 - INFO - ci_test]: all database: ['default', 'prod'] (test_data_persistence.py:26)

[2024-03-04T19:56:12.726Z] [2024-03-04 19:56:12 - INFO - ci_test]: [test][2024-03-04T19:56:12Z] [0.51473979s] Hello_Milvus flush -> None (wrapper.py:30)

[2024-03-04T19:56:12.726Z] [2024-03-04 19:56:12 - INFO - ci_test]: assert create collection: 0.010502099990844727, init_entities: 4000 (test_data_persistence.py:38)

[2024-03-04T19:56:13.693Z] [2024-03-04 19:56:13 - INFO - ci_test]: [test][2024-03-04T19:56:12Z] [0.87494928s] Hello_Milvus insert -> (insert count: 2000, delete count: 0, upsert count: 0, timestamp: 448156709185650702, success count: 2000, err count: 0) (wrapper.py:30)

[2024-03-04T19:56:13.694Z] [2024-03-04 19:56:13 - INFO - ci_test]: assert insert: 0.8752250671386719 (test_data_persistence.py:45)

[2024-03-04T19:56:17.869Z] [2024-03-04 19:56:17 - INFO - ci_test]: [test][2024-03-04T19:56:13Z] [3.61885718s] Hello_Milvus flush -> None (wrapper.py:30)

[2024-03-04T19:56:17.870Z] [2024-03-04 19:56:17 - INFO - ci_test]: [test][2024-03-04T19:56:17Z] [0.51756654s] Hello_Milvus flush -> None (wrapper.py:30)

[2024-03-04T19:56:18.429Z] [2024-03-04 19:56:18 - INFO - ci_test]: [test][2024-03-04T19:56:17Z] [0.51304610s] Hello_Milvus flush -> None (wrapper.py:30)

[2024-03-04T19:56:18.429Z] [2024-03-04 19:56:18 - INFO - ci_test]: assert flush: 4.139860153198242, entities: 6000 (test_data_persistence.py:55)

[2024-03-04T19:56:18.429Z] [2024-03-04 19:56:18 - INFO - ci_test]: index info: [{'collection': 'Hello_Milvus', 'field': 'float_vector', 'index_name': 'test_mcA55UlW', 'index_param': {'index_type': 'HNSW', 'metric_type': 'L2', 'params': {'M': 48, 'efConstruction': 500}}}, {'collection': 'Hello_Milvus', 'field': 'varchar', 'index_name': 'test_fcBufvlI', 'index_param': {}}] (test_data_persistence.py:74)

[2024-03-04T19:56:18.429Z] [2024-03-04 19:56:18 - INFO - ci_test]: [test][2024-03-04T19:56:18Z] [0.01447583s] Hello_Milvus load -> None (wrapper.py:30)

[2024-03-04T19:56:18.989Z] [2024-03-04 19:56:18 - ERROR - pymilvus.decorators]: RPC error: [search], <MilvusException: (code=65535, message=failed to get delegator 1 for channel by-dev-rootcoord-dml_1_448156144239779644v0: can not find client of node 1)>, <Time:{'RPC start': '2024-03-04 19:56:18.230624', 'RPC error': '2024-03-04 19:56:18.840989'}> (decorators.py:134)

[2024-03-04T19:56:18.989Z] [2024-03-04 19:56:18 - ERROR - ci_test]: Traceback (most recent call last):

[2024-03-04T19:56:18.989Z]   File "/home/jenkins/agent/workspace/tests/python_client/utils/api_request.py", line 32, in inner_wrapper

[2024-03-04T19:56:18.989Z]     res = func(*args, **_kwargs)

[2024-03-04T19:56:18.989Z]   File "/home/jenkins/agent/workspace/tests/python_client/utils/api_request.py", line 63, in api_request

[2024-03-04T19:56:18.989Z]     return func(*arg, **kwargs)

[2024-03-04T19:56:18.989Z]   File "/usr/local/lib/python3.8/dist-packages/pymilvus/orm/collection.py", line 792, in search

[2024-03-04T19:56:18.989Z]     resp = conn.search(

[2024-03-04T19:56:18.989Z]   File "/usr/local/lib/python3.8/dist-packages/pymilvus/decorators.py", line 135, in handler

[2024-03-04T19:56:18.989Z]     raise e from e

[2024-03-04T19:56:18.989Z]   File "/usr/local/lib/python3.8/dist-packages/pymilvus/decorators.py", line 131, in handler

[2024-03-04T19:56:18.989Z]     return func(*args, **kwargs)

[2024-03-04T19:56:18.989Z]   File "/usr/local/lib/python3.8/dist-packages/pymilvus/decorators.py", line 170, in handler

[2024-03-04T19:56:18.989Z]     return func(self, *args, **kwargs)

[2024-03-04T19:56:18.989Z]   File "/usr/local/lib/python3.8/dist-packages/pymilvus/decorators.py", line 110, in handler

[2024-03-04T19:56:18.989Z]     raise e from e

[2024-03-04T19:56:18.989Z]   File "/usr/local/lib/python3.8/dist-packages/pymilvus/decorators.py", line 74, in handler

[2024-03-04T19:56:18.989Z]     return func(*args, **kwargs)

[2024-03-04T19:56:18.989Z]   File "/usr/local/lib/python3.8/dist-packages/pymilvus/client/grpc_handler.py", line 794, in search

[2024-03-04T19:56:18.989Z]     return self._execute_search(request, timeout, round_decimal=round_decimal, **kwargs)

[2024-03-04T19:56:18.989Z]   File "/usr/local/lib/python3.8/dist-packages/pymilvus/client/grpc_handler.py", line 735, in _execute_search

[2024-03-04T19:56:18.989Z]     raise e from e

[2024-03-04T19:56:18.989Z]   File "/usr/local/lib/python3.8/dist-packages/pymilvus/client/grpc_handler.py", line 728, in _execute_search

[2024-03-04T19:56:18.989Z]     check_status(response.status)

[2024-03-04T19:56:18.989Z]   File "/usr/local/lib/python3.8/dist-packages/pymilvus/client/utils.py", line 60, in check_status

[2024-03-04T19:56:18.989Z]     raise MilvusException(status.code, status.reason, status.error_code)

[2024-03-04T19:56:18.990Z] pymilvus.exceptions.MilvusException: <MilvusException: (code=65535, message=failed to get delegator 1 for channel by-dev-rootcoord-dml_1_448156144239779644v0: can not find client of node 1)>

[2024-03-04T19:56:18.990Z]  (api_request.py:45)
```

```

[2024-03-04T19:59:51.883Z] <name>: Checker__R2WFN5JU

[2024-03-04T19:59:51.883Z] <description>: 

[2024-03-04T19:59:51.883Z] <schema>: {'auto_id': False, 'description': '', 'fields': [{'name': 'int64', 'description': '', 'type': <DataType.INT64: 5>, 'is_primary': True, 'auto_id': False}, {'name': 'float', 'description': '', 'type': <DataType.FLOAT: 10>}......  (api_request.py:37)

[2024-03-04T19:59:51.883Z] [2024-03-04 19:57:02 - DEBUG - ci_test]: (api_request)  : [Collection.flush] args: [], kwargs: {'timeout': 180} (api_request.py:62)

[2024-03-04T19:59:51.883Z] [2024-03-04 19:57:02 - DEBUG - ci_test]: (api_response) : None  (api_request.py:37)

[2024-03-04T19:59:51.883Z] [2024-03-04 19:57:02 - DEBUG - ci_test]: (api_request)  : [Collection.compact] args: [180], kwargs: {} (api_request.py:62)

[2024-03-04T19:59:51.884Z] [2024-03-04 19:57:02 - DEBUG - ci_test]: (api_response) : None  (api_request.py:37)

[2024-03-04T19:59:51.884Z] [2024-03-04 19:58:01 - DEBUG - ci_test]: (api_request)  : [Collection.flush] args: [], kwargs: {'timeout': 180} (api_request.py:62)

[2024-03-04T19:59:51.884Z] [2024-03-04 19:58:02 - DEBUG - ci_test]: (api_response) : None  (api_request.py:37)

[2024-03-04T19:59:51.884Z] [2024-03-04 19:58:02 - INFO - ci_test]: assert create collection: 1.1373822689056396, init_entities: 200145 (test_all_collections_after_chaos.py:49)

[2024-03-04T19:59:51.884Z] [2024-03-04 19:58:03 - DEBUG - ci_test]: (api_request)  : [Collection.insert] args: [[[-3000, -2999, -2998, -2997, -2996, -2995, -2994, -2993, -2992, -2991, -2990, -2989, -2988, -2987, -2986, -2985, -2984, -2983, -2982, -2981, -2980, -2979, -2978, -2977, -2976, -2975, -2974, -2973, -2972, -2971, -2970, -2969, -2968, -2967, -2966, -2965, -2964, -2963, -2962, -2961, -2960, -2959, -29......, kwargs: {'timeout': 180} (api_request.py:62)

[2024-03-04T19:59:51.884Z] [2024-03-04 19:58:04 - DEBUG - ci_test]: (api_response) : (insert count: 2000, delete count: 0, upsert count: 0, timestamp: 448156738349432833, success count: 2000, err count: 0)  (api_request.py:37)

[2024-03-04T19:59:51.884Z] [2024-03-04 19:58:04 - INFO - ci_test]: assert insert: 1.529909372329712 (test_all_collections_after_chaos.py:57)

[2024-03-04T19:59:51.884Z] [2024-03-04 19:58:04 - DEBUG - ci_test]: (api_request)  : [Collection.flush] args: [], kwargs: {'timeout': 180} (api_request.py:62)

[2024-03-04T19:59:51.884Z] [2024-03-04 19:58:08 - DEBUG - ci_test]: (api_response) : None  (api_request.py:37)

[2024-03-04T19:59:51.884Z] [2024-03-04 19:58:08 - DEBUG - ci_test]: (api_request)  : [Collection.flush] args: [], kwargs: {'timeout': 180} (api_request.py:62)

[2024-03-04T19:59:51.884Z] [2024-03-04 19:58:08 - DEBUG - ci_test]: (api_response) : None  (api_request.py:37)

[2024-03-04T19:59:51.884Z] [2024-03-04 19:58:08 - INFO - ci_test]: assert flush: 3.577561378479004, entities: 202145 (test_all_collections_after_chaos.py:67)

[2024-03-04T19:59:51.884Z] [2024-03-04 19:58:08 - INFO - ci_test]: index info: [{'collection': 'Checker__R2WFN5JU', 'field': 'voice_emb', 'index_name': 'voice_emb', 'index_param': {'index_type': 'HNSW', 'metric_type': 'L2', 'params': {'M': 48, 'efConstruction': 500}}}, {'collection': 'Checker__R2WFN5JU', 'field': 'int64', 'index_name': 'int64', 'index_param': {'index_type': 'INVERTED'}}, {'collection': 'Checker__R2WFN5JU', 'field': 'float', 'index_name': 'float', 'index_param': {'index_type': 'INVERTED'}}, {'collection': 'Checker__R2WFN5JU', 'field': 'varchar', 'index_name': 'varchar', 'index_param': {'index_type': 'INVERTED'}}, {'collection': 'Checker__R2WFN5JU', 'field': 'float_vector', 'index_name': 'float_vector', 'index_param': {'index_type': 'HNSW', 'metric_type': 'L2', 'params': {'M': 48, 'efConstruction': 500}}}, {'collection': 'Checker__R2WFN5JU', 'field': 'image_emb', 'index_name': 'image_emb', 'index_param': {'index_type': 'HNSW', 'metric_type': 'L2', 'params': {'M': 48, 'efConstruction': 500}}}, {'collection': 'Checker__R2WFN5JU', 'field': 'text_emb', 'index_name': 'text_emb', 'index_param': {'index_type': 'HNSW', 'metric_type': 'L2', 'params': {'M': 48, 'efConstruction': 500}}}] (test_all_collections_after_chaos.py:71)

[2024-03-04T19:59:51.884Z] [2024-03-04 19:58:08 - INFO - ci_test]: index info: [{'collection': 'Checker__R2WFN5JU', 'field': 'image_emb', 'index_name': 'image_emb', 'index_param': {'index_type': 'HNSW', 'metric_type': 'L2', 'params': {'M': 48, 'efConstruction': 500}}}, {'collection': 'Checker__R2WFN5JU', 'field': 'text_emb', 'index_name': 'text_emb', 'index_param': {'index_type': 'HNSW', 'metric_type': 'L2', 'params': {'M': 48, 'efConstruction': 500}}}, {'collection': 'Checker__R2WFN5JU', 'field': 'voice_emb', 'index_name': 'voice_emb', 'index_param': {'index_type': 'HNSW', 'metric_type': 'L2', 'params': {'M': 48, 'efConstruction': 500}}}, {'collection': 'Checker__R2WFN5JU', 'field': 'int64', 'index_name': 'int64', 'index_param': {'index_type': 'INVERTED'}}, {'collection': 'Checker__R2WFN5JU', 'field': 'float', 'index_name': 'float', 'index_param': {'index_type': 'INVERTED'}}, {'collection': 'Checker__R2WFN5JU', 'field': 'varchar', 'index_name': 'varchar', 'index_param': {'index_type': 'INVERTED'}}, {'collection': 'Checker__R2WFN5JU', 'field': 'float_vector', 'index_name': 'float_vector', 'index_param': {'index_type': 'HNSW', 'metric_type': 'L2', 'params': {'M': 48, 'efConstruction': 500}}}] (test_all_collections_after_chaos.py:86)

[2024-03-04T19:59:51.884Z] [2024-03-04 19:58:08 - DEBUG - ci_test]: (api_request)  : [Collection.load] args: [None, 1, 180], kwargs: {} (api_request.py:62)

[2024-03-04T19:59:51.884Z] [2024-03-04 19:58:08 - DEBUG - ci_test]: (api_response) : None  (api_request.py:37)

[2024-03-04T19:59:51.884Z] [2024-03-04 19:58:08 - DEBUG - ci_test]: (api_request)  : [Collection.search] args: [[[0.08272373625161705, 0.029546511647984112, 0.0734233883812938, 0.05264910342939128, 0.03841701511147735, 0.11634951617825509, 0.09949213253152069, 0.08482242423294482, 0.0010720500247573383, 0.05179128789378865, 0.045402453787672645, 0.1281071730464219, 0.134043830674233, 0.09416274666311382, 0.0......, kwargs: {} (api_request.py:62)

[2024-03-04T19:59:51.884Z] [2024-03-04 19:58:08 - ERROR - pymilvus.decorators]: RPC error: [search], <MilvusException: (code=503, message=failed to search: node offline[node=16]: channel not available[channel=by-dev-rootcoord-dml_7_448156144239980114v0])>, <Time:{'RPC start': '2024-03-04 19:58:08.912411', 'RPC error': '2024-03-04 19:58:08.916069'}> (decorators.py:134)

[2024-03-04T19:59:51.884Z] [2024-03-04 19:58:08 - ERROR - ci_test]: Traceback (most recent call last):

[2024-03-04T19:59:51.884Z]   File "/home/jenkins/agent/workspace/tests/python_client/utils/api_request.py", line 32, in inner_wrapper

[2024-03-04T19:59:51.884Z]     res = func(*args, **_kwargs)

[2024-03-04T19:59:51.884Z]   File "/home/jenkins/agent/workspace/tests/python_client/utils/api_request.py", line 63, in api_request

[2024-03-04T19:59:51.884Z]     return func(*arg, **kwargs)

[2024-03-04T19:59:51.884Z]   File "/usr/local/lib/python3.8/dist-packages/pymilvus/orm/collection.py", line 792, in search

[2024-03-04T19:59:51.884Z]     resp = conn.search(

[2024-03-04T19:59:51.884Z]   File "/usr/local/lib/python3.8/dist-packages/pymilvus/decorators.py", line 135, in handler

[2024-03-04T19:59:51.884Z]     raise e from e

[2024-03-04T19:59:51.884Z]   File "/usr/local/lib/python3.8/dist-packages/pymilvus/decorators.py", line 131, in handler

[2024-03-04T19:59:51.884Z]     return func(*args, **kwargs)

[2024-03-04T19:59:51.884Z]   File "/usr/local/lib/python3.8/dist-packages/pymilvus/decorators.py", line 170, in handler

[2024-03-04T19:59:51.884Z]     return func(self, *args, **kwargs)

[2024-03-04T19:59:51.884Z]   File "/usr/local/lib/python3.8/dist-packages/pymilvus/decorators.py", line 110, in handler

[2024-03-04T19:59:51.884Z]     raise e from e

[2024-03-04T19:59:51.884Z]   File "/usr/local/lib/python3.8/dist-packages/pymilvus/decorators.py", line 74, in handler

[2024-03-04T19:59:51.884Z]     return func(*args, **kwargs)

[2024-03-04T19:59:51.884Z]   File "/usr/local/lib/python3.8/dist-packages/pymilvus/client/grpc_handler.py", line 794, in search

[2024-03-04T19:59:51.884Z]     return self._execute_search(request, timeout, round_decimal=round_decimal, **kwargs)

[2024-03-04T19:59:51.884Z]   File "/usr/local/lib/python3.8/dist-packages/pymilvus/client/grpc_handler.py", line 735, in _execute_search

[2024-03-04T19:59:51.884Z]     raise e from e

[2024-03-04T19:59:51.884Z]   File "/usr/local/lib/python3.8/dist-packages/pymilvus/client/grpc_handler.py", line 728, in _execute_search

[2024-03-04T19:59:51.884Z]     check_status(response.status)

[2024-03-04T19:59:51.884Z]   File "/usr/local/lib/python3.8/dist-packages/pymilvus/client/utils.py", line 60, in check_status

[2024-03-04T19:59:51.884Z]     raise MilvusException(status.code, status.reason, status.error_code)

[2024-03-04T19:59:51.884Z] pymilvus.exceptions.MilvusException: <MilvusException: (code=503, message=failed to search: node offline[node=16]: channel not available[channel=by-dev-rootcoord-dml_7_448156144239980114v0])>
```

### Milvus Log

failed job: https://qa-jenkins.milvus.io/blue/organizations/jenkins/chaos-test-kafka-cron/detail/chaos-test-kafka-cron/12008/pipeline

log:
[artifacts-querynode-pod-kill-12008-server-logs.tar.gz](https://github.com/milvus-io/milvus/files/14490066/artifacts-querynode-pod-kill-12008-server-logs.tar.gz)

**Comment by weiliu1031** [comment]
same issue: #31040  
leader checker can't update segment's load version, which causes leader view keeping unavailable nodes.

**Comment by congqixia** [comment]
Could be related to patch https://github.com/milvus-io/milvus/pull/31040
Could you please help verify?
/assign @zhuwenxing

**Comment by weiliu1031** [comment]
should be fixed by #31277

**Comment by weiliu1031** [comment]
please verify this
